### PR TITLE
[RA2 Ch03 Ch04] Adding container package manager

### LINKED ISSUE #451: [RA2 Core]: Build a consensus about the usage of Helm and document the result
In the discussions of #383 there was no consensus if Helm should be part of the Reference Architecture. The resolution of this issue shall build a consensus on it and document the result.

### PR DESCRIPTION
**Pull request by CsatariGergely**
This change adds discussion about the selection of container package managers
and a recommendation for Helm v3.

Fixes #451

### CODE REVIEW

File: doc/ref_arch/kubernetes/chapters/chapter04.md [hunk 1]
```diff
@@ -53,7 +53,7 @@
 <a name="4.8"></a>
 ## 4.8 Container package managers
 
-> This chapter should describe which container package managers are used in the Reference Architecture. 
+The reference architecture should include [Helm v3](https://v3.helm.sh/).
```
[the remark this answers is not in the document]
Comment (tomkivlin) [reply doc/ref_arch/kubernetes/chapters/chapter04.md]: Feels there are three options:
1. The architecture includes Helm (or another specific package manager) - I think a **should** does allow for operators who choose not to use Helm.
1. The architecture includes a generic capability that could potentially be met by many of the package managers (maybe gives the operator a choice of which to use?)
1. The architecture doesn't include package managers at all and only references the as another consumer of the Kubernetes API

I don't think the second one would really work at this level as we need to be defining standard components to be deployed by an RI for testing, right? In which case it's a question of whether the capability is in or out of scope.

If in, then I think the wording is good.

My opinion is that it's out as it's a consumer of the Kubernetes API and isn't providing any "infrastructure" services or management of the infrastructure services, it is part of the management of the application/software.

File: doc/ref_arch/kubernetes/chapters/chapter03.md [hunk 2]
```diff
@@ -45,7 +45,7 @@
 <a name="3.2.4"></a>
 ### 3.2.4 Container Package Managers
 
-> This chapter shoud describe what type of container package manager (e.g.: Helm, [CNAB](https://cnab.io/)) related considerations 
+To manage complex applications consisting from several pods the reference architecture should provide a container package manager. The package manager shold be able to manage the lifecycle a set of pods and provide a framework to customize a set of parameters for the deployment.
```
[the remark this answers is not in the document]
Comment (tomkivlin) [reply doc/ref_arch/kubernetes/chapters/chapter03.md]: +1

File: doc/ref_arch/kubernetes/chapters/chapter04.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (CsatariGergely) [reply doc/ref_arch/kubernetes/chapters/chapter04.md]: I think the option is in and not optional. 
It effects how the VNF packages are constructed, this is why I think it should be part of the infrastructure. 

I agree that in the Reference Architecture we should not leave this open.

[the remark this answers is not in the document]
Comment (tomkivlin) [reply doc/ref_arch/kubernetes/chapters/chapter04.md]: I suggest we take the question of whether it is in or out to the discussion in #449/#450 and approve this assuming it's in. If we decide it's out we can remove.

[the remark this answers is not in the document]
Comment (ulikleber) [reply doc/ref_arch/kubernetes/chapters/chapter04.md]: In chapter 2, we now have a Should for some package manager, while in chapter 4 we mandate helm v3. 
We have to be consistent.
I would prefer to be optional, at least for the selected package manager.

**Comment by rabiabdel** [comment]
@msidana @ulikleber please approve.

File: doc/ref_arch/kubernetes/chapters/chapter04.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (CsatariGergely) [reply doc/ref_arch/kubernetes/chapters/chapter04.md]: Optional support in the infra means, that all VNF-s should be prepared for both Helm and non Helm options. I would prefer to have it as mandatory and change the should to must in Ch2.

[the remark this answers is not in the document]
Comment (tomkivlin) [reply doc/ref_arch/kubernetes/chapters/chapter04.md]: @CsatariGergely I agree they need to be the same.  I don't quite understand the logic of a VNF being prepared or not - CNTT is about the infrastructure, not the application - the VNFs (and therefore their preparedness) is not in scope.

In terms of what we include, I see Helm docs have a "Kubernetes Distribution Guide" which lists those that are known to work with Helm, in which case I can see us including any requirements / tests that the Helm team have to validate that position (I can't find any though).   I'm still struggling with Helm itself (the client app and the libraries that talk to the Kubernetes API) being included (as per #449).

**Comment by rabiabdel** [comment]
@ulikleber please approve so this can be merged.

File: doc/ref_arch/kubernetes/chapters/chapter04.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ulikleber) [reply doc/ref_arch/kubernetes/chapters/chapter04.md]: I think in the reference architecture, different options should be possible and we need to provide a consistent statement. Therefore I would prefer should for both chapters.

**Comment by ulikleber** [comment]
> @ulikleber please approve so this can be merged.

We need to resolve the inconsistency, see conversation above.

**Comment by tomkivlin** [comment]
From meeting 7/11/19: wording required to explain that Helm (e.g.) may be required as a package manager, and has requirements on the Kubernetes layer.  Also need to explain that Tiller is not included in RA2 and so Helm 3 (k8s API) is implicitly expected.

**Comment by CsatariGergely** [comment]
> 
> 
> From meeting 7/11/19: wording required to explain that Helm (e.g.) may be required as a package manager, and has requirements on the Kubernetes layer. Also need to explain that Tiller is not included in RA2 and so Helm 3 (k8s API) is implicitly expected.

Hopefully done now.

**Comment by CsatariGergely** [comment]
> 
> 
> @CsatariGergely given the discussion and content in #570 and #449 I think we should remove 3.2.4 and 4.8 as the capability is currently being described as not in scope. If you're not comfortable removing them entirely then I think a note that the capabilities is part of a modern / cloud native VNFM and therefore not in scope of the RA would be ok.

I'm more comfortable with the may and the statement that the RA should support the package managers.

**Pull request merged**